src/kademlia/dht_storage.cpp

```diff
@@ -83,17 +84,15 @@ namespace
 	struct dht_immutable_item
 	{
 		// the actual value
-		std::unique_ptr<char[]> value;
+		std::vector<char> value;
```

Comment:
I agree that vector is convenient. there is a type ``buffer`` in libtorrent, which basically is a single pointer and size + the memory management. It could be a reasonable alternative, but it would also couple the dht to the core library more too.